packages/shopping/public/templates.js

```diff
@@ -0,0 +1,221 @@
+'use strict';
+
+const navBarTemplate = `
+      <a id="logo" class="navbar-brand" href="/shoppy.html">Shoppy</a>
+      <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navigationBar" aria-controls="navigationBar" aria-expanded="false" aria-label="Toggle navigation">
+        <span class="navbar-toggler-icon"></span>
+      </button>
+
+      <div class="collapse navbar-collapse" id="navigationBar">
+        <ul class="navbar-nav mr-auto">
+          <li class="nav-item active">
+            <a class="nav-link" href="/shoppy.html">Home <span class="sr-only">(current)</span></a>
+          </li>
+          <li class="nav-item" id="shoppingCartLink" style="display:none">
+            <a class="nav-link" href="javascript:displayShoppingCart()">Shopping Cart<span style="display:none" id="itemsInCart" class="badge badge-notify">0</span></a>
+          </li>
+          <li id="signUp" class="nav-item">
+            <a class="nav-link" href="#" data-toggle="modal" data-target="#signUpModal" tabindex="-1">Sign Up</a>
+          </li>
+
+<div class="modal" id="signUpModal" tabindex="-1" role="dialog">
```

Comment:
nitpick: should this part be indented (for readability)?

Reply:
> an easier way to feed the data we got from APIs into the UI element

Do we want to use a framework only for feeding data to the UI? If we use a framework, we would better be using it for every aspect of the app. If we develop the app using a framework, the app will not be as simple as it is now.

I prefer Vue to Angular and React. How do we decide which one to use? Majority vote? I think insights to front-end development and developer experience in general should count more than majority vote. Now, how do we measure that?

Let's keep the app as-is with focus on integrating LoopBack APIs to a framework-neutral app.

Feeding server data to the UI will look more "natural" once we refactor the code to [use async-await](https://github.com/strongloop/loopback4-example-shopping/issues/478).